docs/source/indexes/all_neighbors.rst

```diff
@@ -0,0 +1,71 @@
+All-neighbors
```

Comment:
I love the docs, but all-neighbros isn't really an index type, it's more-so a way to take a series of dense vectors and construct a knn graph from them (it's really a way to convert dense vectors into a corresponding knn graph when you think about it).